developer-workflow/extension-modules.rst

```diff
@@ -5,13 +5,579 @@
+platforms.
+
+Updating :cpy-file:`configure.ac`
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. add section about configuration variable afterwards
+
+* Locate the ``SRCDIRS`` variable and add the following line:
+
+  .. code-block:: text
+     :caption: :cpy-file:`configure.ac`
+     :emphasize-lines: 4
+
+     AC_SUBST([SRCDIRS])
+     SRCDIRS="\
+     ...
+     Modules/_foo \
+     ..."
+
+  .. note::
+
+     This step is only needed when adding new source directories to
+     the CPython project.
+
+* Find the section containing ``PY_STDLIB_MOD`` and ``PY_STDLIB_MOD_SIMPLE``
+  usages and add the following line:
+
+  .. code-block:: text
+     :caption: :cpy-file:`configure.ac`
+     :emphasize-lines: 3
+
+     dnl always enabled extension modules
+     ...
+     PY_STDLIB_MOD_SIMPLE([fastfoo], [-I\$(srcdir)/Modules/_foo], [])
+     ...
+
+  The ``PY_STDLIB_MOD_SIMPLE`` macro takes as arguments:
+
+  * the module name as specified by :c:member:`PyModuleDef.m_name`,
+  * the compiler flags (CFLAGS), and
+  * the linker flags (LDFLAGS).
+
+  If the extension module may not be enabled or supported depending on the
+  host configuration, use the ``PY_STDLIB_MOD`` macro instead, which takes
+  as arguments:
+
+  * the module name as specified by :c:member:`PyModuleDef.m_name`,
+  * a boolean indicating whether the extension is **enabled** or not,
+  * a boolean indicating whether the extension is **supported** or not,
+  * the compiler flags (CFLAGS), and
+  * the linker flags (LDFLAGS).
+
+  For instance, enabling the ``fastfoo`` extension on Linux platforms, but
+  only providing support for 32-bit architecture, is achieved as follows:
+
+  .. code-block:: text
+     :caption: :cpy-file:`configure.ac`
+     :emphasize-lines: 2, 3
+
+     PY_STDLIB_MOD([fastfoo],
+                   [test "$ac_sys_system" = "Linux"],
+                   [test "$ARCH_RUN_32BIT" = "true"],
+                   [-I\$(srcdir)/Modules/_foo], [])
+
+  More generally, the status of the extension is determined as follows:
+
+  +-----------+-----------------+----------+
+  | Enabled   | Supported       | Status   |
+  +===========+=================+==========+
+  | true      | true            | yes      |
+  +-----------+-----------------+----------+
+  | true      | false           | missing  |
+  +-----------+-----------------+----------+
+  | false     | true or false   | disabled |
+  +-----------+-----------------+----------+
+
+  The extension status is ``n/a`` if the extension is marked unavailable
+  by the ``PY_STDLIB_MOD_SET_NA`` macro. To mark an extension as unavailable,
+  find the usages of ``PY_STDLIB_MOD_SET_NA`` in :cpy-file:`configure.ac` and
+  add the following line:
+
+  .. code-block:: text
+     :caption: :cpy-file:`configure.ac`
+     :emphasize-lines: 4
+
+     dnl Modules that are not available on some platforms
+     AS_CASE([$ac_sys_system],
+         ...
+         [PLATFORM_NAME], [PY_STDLIB_MOD_SET_NA([fastfoo])],
+         ...
+     )
+
```

Comment:
Likely worth stating explicitly that folks should look at the comments and existing modules in `configure.ac` for guidance on adding new external build dependencies for modules that need them.